### PR TITLE
Create fully working incident update task (old post-importer)

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR integrates the new post-importer fully into the project. There is some duplication around the repositories:

- We create more than one ES/OS client
- This cascades into keeping track of them in the background task manager, so we can close them
- Some test duplication

In the future, one of the tasks we want is to abstract the client itself away to have a single repository implementation for the archiver, and a single implementation for the incident update task. At the same time, we only have to track one closeable resource in the task manager: the client. So there'll be a lot of stuff cut down later, but I think it's OK to accept this as is for now.

I added only basic E2E tests, as more will be covered as part of https://github.com/camunda/camunda/issues/25595. Here, I mostly wanted to add tests for the very basic functionality to make sure it's working as intended.

One thing: the extension used everywhere (`BrokerWithCamundaExporterITInvocationProvider`) is not actually wiring OS. I'll follow up on that with the rest of the project team.

## Related issues

related to #24932

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java [hunk 1]
```diff
@@ -29,12 +31,16 @@ public final class BackgroundTaskManager implements CloseableSilently {
...
+      final @WillCloseWhenClosed IncidentUpdateRepository incidentRepository,
       final Logger logger,
       final @WillCloseWhenClosed ScheduledThreadPoolExecutor executor,
       final List<Runnable> tasks) {
     this.partitionId = partitionId;
-    this.repository = Objects.requireNonNull(repository, "must specify a repository");
+    this.archiverRepository =
+        Objects.requireNonNull(archiverRepository, "must specify an archiver repository");
+    this.incidentRepository =
+        Objects.requireNonNull(incidentRepository, "must specify an archiver repository");
```
Comment (entangled90) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java:43]: :x: "must specify an incident repository"

**Review by entangled90 — APPROVED** [review]
Great! just fix the log message before merging

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java (hunk 1 above)
Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java:43]: :see_no_evil:

Comment (npepinpe) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java:43]:
```suggestion
        Objects.requireNonNull(incidentRepository, "must specify an incident repository");
```

**Pull request merged**